### DISCUSSION TITLE
Не подключается история с last.fm

### DISCUSSION THREAD

**Discussion by igogold** [Chimildic/goofy#148] [Помощь с ошибкой] [answered]
Впервые скопировал себе Goofy (v1.5.3), настроил доступы по инструкции, но периодическое обновление истории прослушивания ругается при запросе к last.fm:

Ошибка
URL: http://ws.audioscrobbler.com/2.0/?method=user.getrecenttracks&user=MyUsername&limit=200&page=1&api_key=MyApiKey&format=json Code: 403 Content: {"message":"Login: User required to be logged in","error":17}

Что ещё за "User required to be logged in"?
Не знаете, случайно, чего хочет last.fm, и как ему в этом помочь?

**Comment by Chimildic** [comment, marked as answer]
Настройка приватности last.fm скрывает историю прослушиваний от всех кроме вас. 

Текущая реализация goofy не позволяет пройти авторизацию, чтобы отправлять такие запросы. Принцип работы можно понимать так: что видно в режиме инкогнито без авторизации, то и может получить goofy. 

Либо снимите приватность в [настройках](https://www.last.fm/settings/privacy) `hide recent listening information`, либо не выйдет получать историю оттуда в данный момент.

**Reply by igogold** [reply]
Спасибо, действительно, у меня в профиле был включён запрет показа недавних треков. После его отключения история стала загружаться без ошибок.
